parteh/PRTAllometricCarbonMod.F90

```diff
@@ -0,0 +1,981 @@
+  ! to include it with the integrator array.
+
+  integer, parameter :: dbh_id             = 7  ! This is just used for the integrator
+  integer, parameter :: n_integration_vars = 7
+
+
+  ! -------------------------------------------------------------------------------------
+  ! Boundary Conditions
+  ! -------------------------------------------------------------------------------------
+  ! Input Boundary Indices (These are public, and therefore
+  !       each boundary condition across all modules must
+  !       have a unique name !!!!)
+  ! -------------------------------------------------------------------------------------
+
+  integer, public, parameter :: ac_bc_inout_id_dbh   = 1   ! Plant DBH
+  integer, public, parameter :: ac_bc_inout_id_netdc = 2   ! Index for the net daily C input BC
+  integer, parameter         :: num_bc_inout         = 2
+
+
+  integer, public, parameter :: ac_bc_in_id_pft   = 1   ! Index for the PFT input BC
+  integer, public, parameter :: ac_bc_in_id_ctrim = 2   ! Index for the canopy trim function
+  integer, parameter         :: num_bc_in         = 2
+
+  ! THere are no purely output boundary conditions
+  integer, parameter         :: num_bc_out      = 0
+
+  ! -------------------------------------------------------------------------------------
+  ! Define the size of the coorindate vector.  For this hypothesis, there is only
+  ! one pool per each species x organ combination.
+  ! -------------------------------------------------------------------------------------
+  integer, parameter         :: icd               = 1   ! Only 1 coordinate per variable
+
+
+  ! -------------------------------------------------------------------------------------
+  ! This is the core type that holds this specific
+  ! plant reactive transport (PRT) module
+  ! -------------------------------------------------------------------------------------
+
+
+  type, public, extends(prt_vartypes) :: callom_prt_vartypes
+
+   contains
+
+     procedure :: DailyPRT     => DailyPRTAC
+     procedure :: FastPRT      => FastPRTAC
+
+   end type callom_prt_vartypes
+   
+   ! ------------------------------------------------------------------------------------
+   !
+   ! This next class is an extention of the base instance that maps state variables
+   !      to the outside model.
+   !
+   ! ------------------------------------------------------------------------------------
+   
+   character(len=*), parameter, private :: sourcefile = __FILE__
+
+
+   ! This is the instance of the mapping table and variable definitions
+   ! this is only allocated once per node
+   class(prt_instance_type), public, target, allocatable :: prt_instance_ac
+
+
+   public :: InitPRTInstanceAC
+
+
+contains
+  
+ 
+  subroutine InitPRTInstanceAC()
+
+     ! ----------------------------------------------------------------------------------
+     ! Initialize and populate the general mapping table that 
```

Comment:
which variable is the mapping table? Not sure I understand what sorts of pre-ordained groups are the context here.